### PR TITLE
Require an unsafe fn or block when dereferencing raw pointers

### PR DESCRIPTION
**Pull request by kumbayo**

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt [hunk 1]
```diff
@@ -89,29 +90,48 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
...
+            annotation.registerFix(AddUnsafeFix(block))
+        }
+    }
+
     private fun checkUnaryExpr(holder: AnnotationHolder, unaryExpr: RsUnaryExpr) {
         val expr = unaryExpr.expr ?: return
         if (unaryExpr.operatorType == UnaryOperator.REF_MUT && !expr.isMutable()) {
             createImmutableErrorAnnotation(holder, expr)
+        } else if (unaryExpr.operatorType == UnaryOperator.DEREF) {
+            checkUnsafePtrDereference(holder, unaryExpr)
```
Comment (farodin91) [inline src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt]: Could you add a quick fix

Comment (kumbayo) [inline src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt]: There is already a SurroundWithUnsafeFix and a AddUnsafeFix being added on line 123-125.
Or am i misunderstanding your request?

Comment (farodin91) [inline src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt]: Oh yes

**Comment by matklad** [comment]
Thanks!

bors r+

**Pull request merged**